lib/executor/constant_arrival_rate.go

```diff
@@ -232,10 +232,17 @@ func (car ConstantArrivalRate) Run(ctx context.Context, out chan<- stats.SampleC
 		}
 	}()
 
+	conf := car.GetConfig()
+	execFn := conf.GetExec().ValueOrZero()
+	env := conf.GetEnv()
+	tags := conf.GetTags()
```

Comment:
This is a bit unsightly, especially since it's repeated in each executor :-/

Should we have a helper function here instead? Pass in config, get `VUActivationParams` back?

Reply:
Yeah, a helper function sounds very, very good. For speed, it might be something like this `func getActivationParams(context.Context, executor.BaseConfig, deactivateCallback func()) *lib.VUActivationParams`, so you don't mess around with slow (i.e. dynamic dispatch) interfaces.

Reply:
See f7e0214c. I refactored `DeactivateCallback()` to take an `InitializedVU`, so that this could be called only once (without needing a closure), but then it leads to [mess like this](https://github.com/loadimpact/k6/blob/f7e0214cc0238c42d1d33c570e673b75cd016813/lib/executor/shared_iterations.go#L228-L229) because of the new context. I suppose it can be called from the same `handleVU` lambda to avoid this, but honestly after the last commit the helper function is not really helping anything :smile: So it might be best to get rid of it after all, and fix this cleanly in the upcoming "executor VU handling" refactor.